### PR TITLE
Add Protobuf 'any' type support to Kafka connector

### PR DESCRIPTION
**Pull request by adamjshook**
## Description

This adds support for the `any` Protobuf data type, converting the message type to JSON. The default implementation uses the `ProtobufUtils` mechanism for parsing the `Descriptor`. There is a separate implementation that uses Confluent's `ProtobufSchemaProvider` which is used instead when the `ConfluentModule` is enabled and uses caching when accessing the types.

This PR needs user-facing documentation which will be added as another commit.

## Additional context and related issues

The Protobuf `any` type contains two fields, one `type_url` field and another being any serialized Protobuf message. The `type_url` is expected to be a URL that resolves to a serialized descriptor of the message. This change uses a `java.net.URL` of the `type_url` to open a stream and read the bytes of the descriptor which is then converted into a `Descriptor` object. We then pass this to a JSON printer to serialize the message. Afterwards, we have to sort all of the keys, including nested keys, as the Trino JSON type expects all keys to be ordered.

This requires users to ensure that the URL is resolvable from the Coordinator and Worker hosts, either by being a locally installed `file://` on each machine or a remote `https://` service.

Expands on #16836 which added `oneof` support

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(X) Release notes are required, with the following suggested text:

```markdown
# Kafka Connector
* Added support for `Any` Protobuf types for file-based and Confluent table descriptors
```

### CODE REVIEW

File: plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/schema/confluent/ConfluentSchemaRegistryConfig.java [hunk 1]
```diff
@@ -40,6 +41,8 @@
     private int confluentSchemaRegistryClientCacheSize = 1000;
     private EmptyFieldStrategy emptyFieldStrategy = IGNORE;
     private Duration confluentSubjectsCacheRefreshInterval = new Duration(1, SECONDS);
+    private int confluentDescriptorCacheSize = 1000;
+    private Duration confluentDescriptorCacheRefreshInterval = new Duration(5, MINUTES);
```
Comment (hashhar) [inline plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/schema/confluent/ConfluentSchemaRegistryConfig.java:45]: hmmm, IMO the cache shouldn't refresh automatically by default

**Review by hashhar — COMMENTED** [review]
lgtm % cache auto-refresh behaviour

Also can you implement a config toggle without which `any` types aren't handled like this? default config must not allow remote schema deserialization by default.

File: plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/schema/confluent/ConfluentSchemaRegistryConfig.java (hunk 1 above)
Comment (adamjshook) [reply plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/schema/confluent/ConfluentSchemaRegistryConfig.java]: Updated to remove a configurable cache and instead use a non-evictable cache with a max size of 1000.

**Comment by adamjshook** [comment]
@mosabua Could you please review the documentation for this type support? Thank you!

**Review by mosabua — CHANGES_REQUESTED** [review]
Docs need some clarification and refinement. Great start and thanks for not ignoring the docs @adamjshook

File: docs/src/main/sphinx/connector/kafka.rst [hunk 2]
```diff
@@ -1430,8 +1431,34 @@ Trino data type                       Allowed Protobuf data type
...
+
+    syntax = "proto3";
+
+    import "google/protobuf/any.proto";
+
+    message schema {
+        google.protobuf.Any any_message = 1;
+    }
+
+The corresponding Trino row is a ``JSON`` field ``any_message``
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/connector/kafka.rst:1459]: cool

**Comment by adamjshook** [comment]
@mosabua I've made updates to the documentation, please give it another review and let me know of any changes. Thank you!

File: docs/src/main/sphinx/connector/kafka.rst [hunk 3]
```diff
@@ -1430,8 +1432,61 @@ Trino data type                       Allowed Protobuf data type
...
+
+    syntax = "proto3";
+
+    import "google/protobuf/any.proto";
+
+    message schema {
+        google.protobuf.Any any_message = 1;
+    }
+
+The corresponding Trino column will be named ``any_message`` of type ``JSON``
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/kafka.rst:1478]:
```suggestion
The corresponding Trino column is named ``any_message`` of type ``JSON``
```

**Review by mosabua — APPROVED** [review]
One last nit, but essentially the docs are good to go.

**Comment by adamjshook** [comment]
@hashhar I've rebased against master to pick up the changes from #17858.

File: docs/src/main/sphinx/connector/kafka.rst [hunk 4]
```diff
@@ -1430,8 +1432,61 @@ Trino data type                       Allowed Protobuf data type
...
 ``TIMESTAMP``                         ``Timestamp``, predefined in ``timestamp.proto``
+``JSON``                              ``oneof`` (Confluent table supplier only), ``Any``
 ===================================== =======================================
 
+any
++++
+
+Message types with an `Any <https://protobuf.dev/programming-guides/proto3/#any>`_
+field contain an arbitrary serialized message as bytes and a type URL to resolve
+that message's type, typically with a scheme of ``file://`` or ``http://``.
```
[the remark this answers is not in the document]
Comment (adamjshook) [reply docs/src/main/sphinx/connector/kafka.rst]: No, I've changed the docs to be more clear.

**Comment by hashhar** [comment]
Rebased to resolve conflicts. Will merge after CI.

**Review by hashhar — APPROVED** [review]

**Pull request merged**